swap_meet/vendor.py

```diff
@@ -1,2 +1,108 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        """
+        Input: item.
+        Result: appends item to inventory attribute, a list in class Vendor.
+        Also returns item. If item doesn't exist returns False.
+        """
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        """
+        Input: item.
+        Result: removes item from inventory attribute, a list in class Vendor.
+        Also returns item. If item doesn't exist returns False.
+        """
+        if item not in self.inventory:
+            return False
+
+        self.inventory.remove(item)
+        
+        return item
+
+    def get_by_category(self, a_category):
+        """
+        Input: category of possible vendor items
+        Results/Output: returns list of items with the input category
+        """
+        items = []
+        for item in self.inventory:
+            if item.category == a_category:
+                items.append(item)
+        return items
+    
+    def swap_items(self, friend, my_item, their_item):
+        """
+        Input: Second vendor, self's item, second vendors item.
+
+        Results/Output: Return false if either item is not in the
+        associated vendors inventory. Returns true and switches items
+        if both vendors have the items input.
+        """
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False
+        else:    
+            self.remove(my_item)
+            friend.add(my_item)
+            
+            friend.remove(their_item)
+            self.add(their_item)
+            return True
+
+    def swap_first_item(self, friend):
+        """
+        Input: Second vendor.
+        Results/Output: if both vendors have item(s) in inventory,
+        they swap first items in their lists of inventories
+        """
+        if len(self.inventory) != 0 and len(friend.inventory) != 0:
```

Comment:
If either inventory is empty, we're supposed to return False, it looks like that isn't happening here. Our tests still pass because it checks if the result is falsey. Functions return None by default if there is no return value, which is falsey. However, the instructions specify to return False.